### ISSUE TITLE
OSL dataset generation over MCP, and an inert --seed-start under deterministic settings

### ISSUE THREAD

**Issue by a-attia** [dmaldona/uqgrid#57]
## Context

AutoHype consumes UQGrid output for forced-oscillation source localization. We
provision the server at a pinned commit (currently `1cf4125` on
`feature/uqgrid-mcp-server`) and drive it through an MCP client. Two things came
up while wiring that up. The first is a feature request; the second is a
correctness note that may matter beyond us.

## 1. OSL dataset generation is not reachable over MCP

`feature/uqgrid-mcp-server` exposes a job-oriented API -- `submit_dynamics`,
`get_job`, `query_result` -- and `submit_dynamics` runs bus-fault scenarios.
`uqgrid.osl` exists on the branch as a library, and `scripts/osl/generate_dataset.py`
drives it, but no MCP tool reaches either. Generating an OSL dataset therefore
means shelling out to the script in the provisioned checkout rather than going
through the service.

That works, and we do it today, but it means dataset generation sits outside the
job/artifact model: no job record, no `result_id`, no artifact download, and no
way for a remote client to request a dataset at all once the server is hosted
rather than local.

Would you consider exposing OSL generation as a tool -- something like
`submit_osl_dataset(fo_buses, freqs, amplitudes, ...) -> Job` returning a normal
job whose result carries the `.npz`/`.json`/manifest artifacts? If that is
already planned, we are happy to wait and keep using the script; knowing the
intended direction is enough for us to avoid building something that gets
replaced.

For reference, the earlier `feature-api--mcp` branch exposed `generate_osl_dataset`
directly, so the capability existed at one point. We deliberately pinned the
newer job-oriented branch instead, since that is where development is active.

## 2. `--seed-start` has no effect under common settings

This one is worth checking independently of the feature request.

`generate_dataset.py` passes `scenario["seed"]` to exactly two places: the
`ColoredNoise` constructor, and `PMUEmulator(seed=...)`. Within `PMUEmulator`
the seed only drives the P-class/M-class assignment when `p_class_fraction` is
strictly between 0 and 1, and the dropout mask when `missing_rate > 0`.

So with `--no-noise --p-class-fraction 1.0 --missing-rate 0.0`, nothing
stochastic remains and the seed is inert. The label and the JSON sidecar still
record it, which makes two runs look like different realizations when the arrays
are bit-identical.

Reproduced on `1cf4125` with the in-repo IEEE 9-bus case:

```bash
# identical arrays, different recorded seeds
generate_dataset.py --raw data/ieee9_v33_mod1.raw --dyr data/ieee9bus_gov.dyr \
  --fo-buses 1 --freqs 0.8 --amplitudes 0.2 --tend 4.0 \
  --no-noise --missing-rate 0.0 --p-class-fraction 1.0 --seed-start 1000
# ...and again with --seed-start 7777  ->  every array byte-identical

# leaving p_class_fraction at the config default of 0.7 instead:
#   V_mag, V_ang, I_mag, I_ang, pmu_class, filter_cutoff_hz_per_bus all differ
```

The behaviour is arguably correct -- a deterministic pipeline should be
deterministic -- but the recorded seed implies variability that is not there.
Two possibilities, whichever you prefer:

- warn when a seed is recorded but no stochastic component is active; or
- document that `seed` is meaningful only with colored noise, partial P-class
  coverage, or a non-zero missing rate.

We hit this while trying to regenerate an independent training set by varying
the seed, which quietly produced the same data. Not a UQGrid bug in the strict
sense, but it cost us a while to find, and anyone using the seed as a
realization index would hit it too.

Happy to open a PR for either item if that is useful.
